diff -crN a/src/Mayaqua/Mayaqua.h b/src/Mayaqua/Mayaqua.h
*** a/src/Mayaqua/Mayaqua.h	Sat Jan  4 10:31:38 2014
--- b/src/Mayaqua/Mayaqua.h	Sun Jan  5 20:18:35 2014
***************
*** 252,268 ****
  #include <ifaddrs.h>
  #endif	// MAYAQUA_SUPPORTS_GETIFADDRS
  
! #ifdef	UNIX_LINUX
! typedef void *iconv_t;
! iconv_t iconv_open (__const char *__tocode, __const char *__fromcode);
! size_t iconv (iconv_t __cd, char **__restrict __inbuf,
!                      size_t *__restrict __inbytesleft,
!                      char **__restrict __outbuf,
!                      size_t *__restrict __outbytesleft);
! int iconv_close (iconv_t __cd);
! #else	// UNIX_LINUX
  #include <iconv.h>
! #endif	// UNIX_LINUX
  
  
  
--- 252,268 ----
  #include <ifaddrs.h>
  #endif	// MAYAQUA_SUPPORTS_GETIFADDRS
  
! //#ifdef	UNIX_LINUX
! //typedef void *iconv_t;
! //iconv_t iconv_open (__const char *__tocode, __const char *__fromcode);
! //size_t iconv (iconv_t __cd, char **__restrict __inbuf,
! //                     size_t *__restrict __inbytesleft,
! //                     char **__restrict __outbuf,
! //                     size_t *__restrict __outbytesleft);
! //int iconv_close (iconv_t __cd);
! //#else	// UNIX_LINUX
  #include <iconv.h>
! //#endif	// UNIX_LINUX
  
  
  
