diff -crN a/rc/vpnbridge b/rc/vpnbridge
*** a/rc/vpnbridge	Thu Jan  1 00:00:00 1970
--- b/rc/vpnbridge	Tue Jan  7 14:07:54 2014
***************
*** 0 ****
--- 1,10 ----
+ #!/bin/sh /etc/rc.common
+ START=60
+ 
+ start(){
+ 	/usr/bin/env LANG=en_US.UTF-8 /usr/bin/vpnbridge start
+ }
+ 
+ stop(){
+ 	/usr/bin/env LANG=en_US.UTF-8 /usr/bin/vpnbridge stop
+ }
diff -crN a/rc/vpnserver b/rc/vpnserver
*** a/rc/vpnserver	Thu Jan  1 00:00:00 1970
--- b/rc/vpnserver	Tue Jan  7 14:07:54 2014
***************
*** 0 ****
--- 1,10 ----
+ #!/bin/sh /etc/rc.common
+ START=60
+ 
+ start(){
+ 	/usr/bin/env LANG=en_US.UTF-8 /usr/bin/vpnserver start
+ }
+ 
+ stop(){
+ 	/usr/bin/env LANG=en_US.UTF-8 /usr/bin/vpnserver stop
+ }
